plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -798,4 +792,62 @@ export class EventsProcessor {
             )
         }
     }
+
+    private async handleTablesDependingOnPersonID(
+        sourcePerson: Person,
+        targetPerson: Person,
+        client?: PoolClient
+    ): Promise<undefined> {
+        // When personIDs change, update places depending on a person_id foreign key
+
+        // For Cohorts
+        await this.db.postgresQuery(
+            'UPDATE posthog_cohortpeople SET person_id = $1 WHERE person_id = $2',
+            [targetPerson.id, sourcePerson.id],
+            'updateCohortPeople',
+            client
+        )
+
+        // For FeatureFlagHashKeyOverrides
+        const allOverrides = await this.db.postgresQuery(
+            'SELECT id, person_id, feature_flag_key FROM posthog_featureflaghashkeyoverride WHERE team_id = $1 AND person_id = ANY($2)',
```

Comment:
(didn't look further into this PR): Why do we select + do post-processing + update instead of a single update query? We're introducing a lot of new failure modes here.

Reply:
A https://stackoverflow.com/questions/5874453/postgresql-way-to-insert-row-with-on-conflict-clause-semantics might work better if you're looking for UPDATE but avoids conflicts case. - just add `ON CONFLICT DO NOTHING` to the INSERT.